src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1118,6 +1154,22 @@ private function parse_next_tag() {
 				continue;
 			}
 
+			/*
+			 * If a non-alpha starts the tag name in a tag closer it's a comment.
+			 * Find the first `>`, which closes the comment.
+			 *
+			 * See https://github.com/WordPress/wordpress-develop/pull/4256
```

Comment:
Would it make sense to link to the spec as a somewhat more canonical source instead? While it's fair that this PR states the intent behind this change, I've been itching to see _why_ we can actually do this 😅 

How about e.g.

```suggestion
			 * See https://html.spec.whatwg.org/#parse-error-invalid-first-character-of-tag-name
```

(which is the reference we're also using [in the tests](https://github.com/WordPress/wordpress-develop/pull/4256/files#diff-f2dca46bdddb8b5f5c56051017e0d0ccd262b0401ba7bdb9db506187a40220b3R1672))

Reply:
good catch; this was probably me pasting the wrong thing and not noticing it

Reply:
updated in 663055b55f